tests/manage/z_cluster/nodes/test_nodes_restart_ms.py

```diff
@@ -293,5 +299,71 @@ def test_node_maintenance_restart(self, nodes, pvc_factory, pod_factory):
                 cephcluster_yaml["status"]["phase"] == expected_phase
             ), f"Status of cephcluster {cephcluster_yaml['metadata']['name']} is {cephcluster_yaml['status']['phase']}"
 
+        verify_provider_osd_nodes_on_correct_machine_pools()
         # Create PVCs and pods
         self.sanity_helpers.create_resources_on_ms_consumers()
+
+    @polarion_id("OCS-3982")
+    def test_provider_worker_nodes_restart(self, nodes):
```

Comment:
wouldn't this test be applicable to non MS setups?

Reply:
No. Because in the MS provider cluster, we have two sets of nodes: One for osd nodes and one for ceph nodes.
Which is different than non-MS clusters.
For non-MS clusters, we already have the node restart test.